beetsplug/substitute.py

```diff
@@ -34,8 +35,9 @@ def tmpl_substitute(self, text):
         """Do the actual replacing."""
         if text:
             for pattern, replacement in self.substitute_rules:
-                if pattern.match(text.lower()):
-                    return replacement
+                new_text, subs_made = re.subn(pattern, replacement, text)
```

Comment:
You may want to use the compiled pattern here
```suggestion
                new_text, subs_made = pattern.subn(replacement, text)
```